mantle/kola/tests/upgrade/basic.go

```diff
@@ -313,10 +313,31 @@ func runFnAndWaitForRebootIntoVersion(c cluster.TestCluster, m platform.Machine,
 	}
 }
 
+func waitForUpgradeToBeStaged(c cluster.TestCluster, m platform.Machine) {
+       // Here we set up a systemd path unit to watch for when ostree
+       // behind the scenes updates the refs in the repo under the
+       // /ostree/repo/refs/heads/ostree/1/1 directory. refchanged.path
+       // will trigger when it gets updated and will then stop wait.service.
+       // The systemd-run --wait causes it to not return here (and thus
+       // continue execution of code here) until wait.service has been
+       // stopped by refchanged.service. This is an effort to make us
+       // start waiting inside runFnAndWaitForRebootIntoVersion until
+       // later in the upgrade process because we are seeing failures due
+       // to timeouts and we're trying to reduce the variability by
+       // minimizing the wait inside that function to just the actual reboot.
+       // https://github.com/coreos/fedora-coreos-tracker/issues/1805
+       //
+       // Note: if systemd-run ever gains the ability to --wait when
+       //       generating a path unit then the below can be simplified.
+       c.RunCmdSync(m, "sudo systemd-run -u refchanged --path-property=PathChanged=/ostree/repo/refs/heads/ostree/1/1 systemctl stop wait.service")
```

Comment:
The canonical API for this is `/ostree/deploy`, which ostree touches whenever it changed the deployments exactly for the purpose of other things wanting to monitor changes:

https://github.com/ostreedev/ostree/blob/ab8a7f7855b0e0a7f3fe7214b77521268b994ce4/src/libostree/ostree-sysroot.c#L449

Reply:
+1

so we just need to change this to `path-property=PathChanged=/ostree/deploy` then and update the comment accordingly?

Reply:
Yeah, though I wonder if it's even necessary. Once the machine goes down for a reboot, it'll stop the `wait` unit below, so you should already get the desired effect. I guess `RunCmdSync` might mark that as an error though depending on how `systemd-run` exits.